src/ConfirmProvider.js

```diff
@@ -40,36 +40,56 @@ const buildOptions = (defaultOptions, options) => {
 const ConfirmProvider = ({ children, defaultOptions = {} }) => {
   const [options, setOptions] = useState({ ...DEFAULT_OPTIONS, ...defaultOptions });
   const [resolveReject, setResolveReject] = useState([]);
-  const [resolve, reject] = resolveReject;
+  const [resolve, reject] = resolveReject
+  /**
+   * Possible states:
+   * 1. "Idle" initially
+   * 2. "Open" when `confirm` is called
+   * 3. "Submitting" when confimation or cancellation button is pressed.
+   * After submitting, the state is back to "Idle" since the Dialog is closed.
+   */
+  const [state, setState] = useState('idle');
 
   const confirm = useCallback((options = {}) => {
     return new Promise((resolve, reject) => {
       setOptions(buildOptions(defaultOptions, options));
       setResolveReject([resolve, reject]);
+      setState('open')
     });
   }, []);
 
   const handleClose = useCallback(() => {
     setResolveReject([]);
+    setState('idle');
   }, []);
 
   const handleCancel = useCallback(() => {
-    reject();
-    handleClose();
-  }, [reject, handleClose]);
+    if (state === 'open') {
+      reject();
```

Comment:
I think we could just do `if (reject) {}` to achieve the same, without an additional state and effect. wdyt?

Reply:
Yes, I agree that it would solve the issue.

But, the check on `state` variable's value as `open` indicates more clearly that the resolve and reject are already available, so I did not have to check it explicitly.

Also, if this was a typescript project, this line would definitely throw a TS error that reject could be `undefined` 😅

Reply:
Well, checking if we have `reject` is more a more direct way of ensuring it's callable, but I get your point. The advantage is that we wouldn't need the additional useEffect, which makes us re-render.

Reply:
Yep